core/node/state_keeper/src/keeper.rs

```diff
@@ -582,34 +592,55 @@ impl ZkSyncStateKeeper {
                     "L1 batch #{} should be sealed unconditionally as per sealing rules",
                     updates_manager.l1_batch.number
                 );
+
+                // Push the current block if it has not been done yet and this will effectively create a fictive l2 block
+                if updates_manager.has_next_block_params() {
+                    self.io.update_next_l2_block_timestamp(
+                        &mut updates_manager
+                            .get_next_l2_block_params_or_batch_params()
+                            .timestamp,
```

Comment:
This doesn't update the timestamp inside `UpdatesManager` because `get_next_l2_block_params_or_batch_params()` returns a copy of params; see its signature. For it to work properly, you should revert to providing a mutable ref to the timestamp from `UpdatesManager`.